modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectRelationshipLocalServiceTest.java

```diff
@@ -1854,6 +1879,23 @@ public void testUpdateObjectRelationship() throws Exception {
 			_objectDefinitionLocalService, new String[] {"C_A", "C_AA"},
 			_objectEntryLocalService, _objectRelationshipLocalService);
 
+		AssertUtils.assertFailure(
+			ObjectRelationshipEdgeException.class,
+			"Inheritance between modifiable system and custom object " +
+				"definitions is not allowed",
+			() -> _bindObjectDefinitions(
+				ObjectRelationshipTestUtil.addObjectRelationship(
+					_objectRelationshipLocalService,
+					_modifiableSystemObjectDefinition, _objectDefinition1)));
+		AssertUtils.assertFailure(
+			ObjectRelationshipEdgeException.class,
+			"Inheritance between modifiable system and custom object " +
+				"definitions is not allowed",
+			() -> _bindObjectDefinitions(
+				ObjectRelationshipTestUtil.addObjectRelationship(
+					_objectRelationshipLocalService, _objectDefinition1,
+					_modifiableSystemObjectDefinition)));
```

Comment:
I'm unsure if we must keep these asserts closer to the creation of _objectDefinition1.

Reply:
But `_objectDefinition1 ` is created in the `setUp`  method [here](https://github.com/liferay-bpm/liferay-portal/blob/66fe3a418adb10770f39b636777cc4e3f39f593d/modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectRelationshipLocalServiceTest.java#L144-L151)